src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -84,6 +85,33 @@ MonitoredTrip preCreateHook(MonitoredTrip monitoredTrip, Request req) {
         return monitoredTrip;
     }
 
+    /**
+     * Performs the operations/checks common to the preCreate and preUpdate hooks.
+     */
+    private void preCreateOrUpdateChecks(MonitoredTrip monitoredTrip, Request req) {
+        checkTripCanBeMonitored(monitoredTrip, req);
+        checkTripBelongsToManagedUser(monitoredTrip, req);
+        processTripQueryParams(monitoredTrip, req);
+    }
+
+    /**
+     * Checks that the monitored trip to be written belongs to the requesting user
+     * or to an OtpUser managed by the requesting ApiUser.
+     *
+     * For update (PUT) requests, this check is in addition to MonitoredTrip#canBeManagedBy
```

Comment:
That's right, `MonitoredTrip#canBeManagedBy` must be ran twice. Currently it is ran on the existing entity, but it also need to run on the entity to be written. Perhaps the additional check I introduced could be moved to `ApiController` since there are no trip-specific code (besides the comment).

Reply:
Yes, I think you're right: this probably belongs in ApiController.

Reply:
Have updated.